lib/main.dart

```diff
@@ -16,13 +18,67 @@ class MyApp extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     // TODO: Implement routing
-    return MaterialApp(
-      title: 'Flutter Demo',
-      theme: ThemeData(
-        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
-        useMaterial3: true,
+    return BlocProvider<ManageGeoCubit>(
+      create: (BuildContext context) => ManageGeoCubit()..checkPermission(),
+      child: MaterialApp(
+        title: 'Flutter Demo',
+        theme: ThemeData(
+          colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+          useMaterial3: true,
+        ),
+        home: const TestCubit(),
       ),
-      home: const FuelStationListScreen(),
+    );
+  }
+}
+
+class TestCubit extends StatefulWidget {
+  const TestCubit({super.key});
+
+  @override
+  State<TestCubit> createState() => _TestCubitState();
+}
+
+class _TestCubitState extends State<TestCubit> {
+  @override
+  void initState() {
+    context.read<ManageGeoCubit>().fetchPositionStream();
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<ManageGeoCubit, ManageGeoState>(
+      builder: (BuildContext context, ManageGeoState state) {
```

Comment:
Totaly fine to use. But you could also use `ManageGeoCubit cubit = context.watch<ManageGeoCubit>();`
This way we don't need this extra BlocBuilder widget and have a bit more flexibility.